### ISSUE TITLE
environments

### ISSUE THREAD

**Issue by KevinBatdorf** [themosis/themosis#146]
I think the current set up encourages people to version control their .env files. An alternative would be to eliminate the config/environment.php file and load the environment based on an environment variable. 

Such as:

APP_ENV=local

**Comment by jlambe** [comment]
From your `config/environment.php` file, you can also define a closure and look after environment variables and write your own condition to detect the environment:
- http://framework.themosis.com/docs/1.3/environment/#load-your-environment

Refer to the "Identify environment using a server environment variable" section in the documentation for details.

**Comment by KevinBatdorf** [comment]
I understand but it still indirectly encourages the user to version control their environment variables this way. 

"Indirectly" as in the alternative forces the user to upload a separate .env file on each server (version control would break the setup)

**Comment by jlambe** [comment]
I'm not sure to understand your issue. You can also provide only a generic `.env` file or even a `.env.project` file that is common to all environments. Then the `config/environment.php` file handles the logic behind.

Can you illustrate your idea with more details?

**Comment by KevinBatdorf** [comment]
it's not an issue, it's just a suggestion to handle the logic with a single .env file per server, ie removing the ability to have multiple extensions (.env.local, .env.production).

The reasoning is that it allows users to keep multiple environment files on each server and indirectly encourages them to version control their .env files. Especially newer users.

Additionally, I'd say the environment docs are a bit intimidating for newer developers.

So basically, you load a single .env file without extension (as in not .env.local), then you use something like `'/config/environments/' . env(APP_ENV) . '.php'` to load the specific config for the server.

**Comment by jlambe** [comment]
Release 2.0 is now using a single `.env` file. The environment name is defined through a variable `APP_ENV`.

There are no more individual configuration based on the environment name as those were merely duplicates, especially for WordPress configuration.
